src/test/Dashboard/Charts/ChartCard.test.tsx

```diff
@@ -0,0 +1,229 @@
+              </Provider>
+            </ChartContext.Provider>
+          </NotificationsContext.Provider>
+        </ServiceContext.Provider>
+      );
+    });
+    expect(tree.toJSON()).toMatchSnapshot('with-content');
+  });
+
+  it('renders loading state correctly', async () => {
+    jest.spyOn(mockController, 'attach').mockReturnValue(of(ControllerState.UNKNOWN));
+
+    let tree;
+    await act(async () => {
+      tree = renderer.create(
+        <ServiceContext.Provider value={defaultServices}>
+          <NotificationsContext.Provider value={NotificationsInstance}>
+            <ChartContext.Provider value={mockChartContext}>
+              <Provider store={store}>
+                <ChartCard theme={'light'} chartKind={'CPU Load'} duration={120} period={10} span={6} dashboardId={0} />
+              </Provider>
+            </ChartContext.Provider>
+          </NotificationsContext.Provider>
+        </ServiceContext.Provider>
+      );
+    });
+    expect(tree.toJSON()).toMatchSnapshot('loading-view');
+  });
+
+  it('renders empty state correctly', async () => {
+    jest.spyOn(mockController, 'attach').mockReturnValue(of(ControllerState.NO_DATA));
+
+    let tree;
+    await act(async () => {
+      tree = renderer.create(
+        <ServiceContext.Provider value={defaultServices}>
+          <NotificationsContext.Provider value={NotificationsInstance}>
+            <ChartContext.Provider value={mockChartContext}>
+              <Provider store={store}>
+                <ChartCard theme={'light'} chartKind={'CPU Load'} duration={120} period={10} span={6} dashboardId={0} />
+              </Provider>
+            </ChartContext.Provider>
+          </NotificationsContext.Provider>
+        </ServiceContext.Provider>
+      );
+    });
+    expect(tree.toJSON()).toMatchSnapshot('empty-state');
+  });
+
+  it('renders empty state with information and action button', async () => {
+    jest.spyOn(mockController, 'attach').mockReturnValue(of(ControllerState.NO_DATA));
+
+    renderChartCard(
+      <ChartCard theme={'light'} chartKind={'CPU Load'} duration={120} period={10} span={6} dashboardId={0} />
+    );
+
+    expect(screen.getByText('CPU Load')).toBeInTheDocument();
+    expect(screen.getByText('CHART_CARD.NO_RECORDING.TITLE')).toBeInTheDocument();
+    expect(screen.getByText('CHART_CARD.NO_RECORDING.DESCRIPTION')).toBeInTheDocument();
+    expect(screen.getByRole('button', { name: /create/i })).toBeInTheDocument();
+  });
+
+  it('navigates to recording creation with prefilled state when empty state button clicked', async () => {
+    jest.spyOn(mockController, 'attach').mockReturnValue(of(ControllerState.NO_DATA));
+
+    const { user } = renderChartCard(
+      <ChartCard theme={'light'} chartKind={'CPU Load'} duration={120} period={10} span={6} dashboardId={0} />
+    );
+
+    expect(history.location.pathname).toBe('/');
+    await user.click(screen.getByRole('button', { name: /create/i }));
```

Comment:
We should use the localized version of the button text to search? Same for other places.